functions/pupil-status/pupil-status.service.js

```diff
@@ -69,6 +69,33 @@ async function changePupilState (pupilId, targetStatusCode) {
   return sqlService.modify(sql, params)
 }
 
+/**
+ * Filter for live checks and make a request for the pupil-status to be updated for multiple pupils on v2
+ * @param {Object} logger
+ * @param {String} logPrefix
+ * @param {[{checkId: <number>, pupilId: <number>, checkCode: <string>, isLiveCheck: <boolean>}]} checkData - must contain `checkId`, `pupilId`, `checkCode` and `isLiveCheck` props
+ * @return {Promise<*|Promise<*>>}
+ */
+async function updatePupilStatusForLiveChecksV2 (logger, logPrefix, checkData) {
+  logger.info(`${logPrefix}: updatePupilStatusV2(): got ${checkData.length} pupils`)
+  // Batch the async messages up for live checks only, to limit max concurrency
+  const batches = R.splitEvery(100, R.filter(c => c.isLiveCheck, checkData))
+  checkData = null
+
+  batches.forEach(async (checks, batchNumber) => {
+    try {
+      await this.addMessageToQueue('pupil-status', {
+        version: 2,
+        messages: checks
+      })
+      logger.verbose(`${logPrefix}: batch ${batchNumber} complete`)
+    } catch (error) {
+      logger.error(`${logPrefix}: updatePupilStatusV2(): ERROR: ${error.message}`)
```

Comment:
Pupil status requests are idempotent.  But if some part of the system has asked for an update there is surely work to do, so just ignoring the failure will lead to incorrect pupil status's.

I'd suggest backing off and re-sending the message several times until it is sent to the queue.

Reply:
@jon-shipley I believe this needs to be isolated outside of the scope of this work.